xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1165,6 +1170,25 @@ bool CDVDDemuxFFmpeg::SeekTime(double time, bool backwards, double* startpts)
 
   if (m_checkvideo)
   {
+    int elapsedMs = 0;
+    auto started = std::chrono::system_clock::now();
+
+    while (!IsVideoReady() && elapsedMs < 1000)
+    {
+      DemuxPacket *pkt = Read();
+      if (pkt)
+        CDVDDemuxUtils::FreeDemuxPacket(pkt);
+      else
+        Sleep(10);
+      m_pkt.result = -1;
+      av_packet_unref(&m_pkt.pkt);
+
+      elapsedMs = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now() - started).count();
```

Comment:
Kodi has a timer implementation which is much nicer than this
https://github.com/xbmc/xbmc/blob/7a8f084f89c9124931a948614269aca4deae6b1a/xbmc/threads/SystemClock.h#L33

Reply:
Cool, I love that.